### DISCUSSION TITLE
Render HTML document and revealjs presentation

### DISCUSSION THREAD

**Discussion by ShKlinkenberg** [quarto-dev/quarto-cli#1751] [Q&A] [answered]
I'm running into problems trying to generate an html website that would also contain my revealjs presentation.

What is the best way to create multi document output for hlml and reveal.

As both are essentially html and the file names are the same, the files get over written.

**Comment by mcanouil** [comment]
See #1433 especially my answer.

**Reply by ShKlinkenberg** [reply]
Thanks @mcanoui, I implemented something along these lines. Rendered the output from a subdir with it's own _quarto.yml file to a different ../dir and used the dir as resource for the site. It would be awesome have multiple outputs in one yml file.

**Comment by dragonstyle** [comment, marked as answer]
You could also try using `output-file` to provide an alternate name for one of the formats, e.g.

```yaml
format:
  html: default
  revealjs:
    output-file: page-reveal.html
```

HTH!

**Reply by ShKlinkenberg** [reply]
Thank you @dragonstyle for your suggestion. I will try this. Wonder if it works from one yaml file.

**Reply by ShKlinkenberg** [reply]
@dragonstyle it just does not work. Even the minimal example does not render both files. It just renders the final one, not the first. Or it is overwriting. It does allow to select HTML or revealjs form the render button, but it is just not running both.

```r
---
title: "Quarto"
format: 
  html: default
  revealjs: 
    output-file: slide.html
---

## Quarto
```

**Comment by mcanouil** [comment]
It does work but not as you the way you seem to want.
Using output-file to write a set of files, then change the output-file to avoid having your files being overwritten works perfectly.

What you did not say, is that you actually want to generate both at the same time at the same place with the same name.
This will never work, but if you are willing to make some changes, the "profile" pre-release feature might help you.
See <https://quarto.org/docs/projects/profiles.html>.

**Comment by perellonieto** [comment]
I have a problem which may be the same as the one pointed out in this question. Let me know if I should open another question.

I am trying to create a website in which some of the qmd pages need to be rendered in the formats html and revealjs at the same time. When I run quarto without a configuration file the resulting website contains one folder with figures for the html and one for the Reveal JS. However, when I use a configuration file `_quarto.yml` the resulting website is missing the folder with the figures for the Reveal JS. What am I doing wrong?

I try to expose here all the details, but you can check the GitHub repository at the end of this comment with a full example.

`index.qmd`
````
---
title: a title
format:
  html: default
  revealjs:
    output-file: slides-index.html
jupyter: python3
---

## Title

The next figure should appear in both html and revealjs 

```{python}
import matplotlib.pyplot as plt

plt.plot([0, 1], [0, 1])
plt.show()
```

````

If I render the code with

```
quarto render index.qmd
```

The resulting webiste contains a folder with figures for the html and the revealjs versions.

If I add a minimal Quarto configuration file

`_quarto.yml`
```
project:
  output-dir: _site
```

Then, rendering with

```
quarto render index.qmd
```

Generates a website in `_site` with only one folder with figures for the html version, the Revealjs folder is missing.

You can find the full example and explanation in the following GitHub repository https://github.com/perellonieto/quarto_html_revealjs_test/tree/3c54d7966d9a36dbe848e9eca15c109b64d468c2

**EDIT**: Updated the previous url to point to the correct commit (Dec 22)

And the expected rendered website https://www.perellonieto.com/quarto_html_revealjs_test/

**Reply by mcanouil** [reply]
To render a whole project you need to use `quarto render`.
Running `quarto render index.qmd` means to render `index.qmd` and its dependencies, nothing more.

**Reply by perellonieto** [reply]
I am sorry for the confusion, the problem is the same with  `quarto render .` and the YAML file that I mentioned. The resulting website has one html for the webpage and one for the slides, but only one folder with the files of the webpage (the folder for the slides is missing).

```
_site/
├── index_files
├── index.html
└── slides-index.html

1 directory, 2 files
```

P.S. By the way, in the mentioned example the whole project is only one file `index.md`. I am trying to exemplify in which case the two folders are generated, and in which case the folder for the slides is missing. Please check the GitHub repository mentioned at the end of my comment.

**Reply by cderv** [reply]
The files for `revealjs` format are also in `index_files`. `libs` contains `revealjs` folder, and figures is in `figure-revealjs`

* https://github.com/perellonieto/quarto_html_revealjs_test/tree/gh-pages/index_files/figure-revealjs
* https://github.com/perellonieto/quarto_html_revealjs_test/tree/gh-pages/index_files/libs/revealjs

You HTML files point to this folder
```html
<script src="index_files/libs/revealjs/dist/reveal.js"></script>
```
https://github.com/perellonieto/quarto_html_revealjs_test/blob/80069b4875a7439d87eac49149c836218a7bdf7c/slides-index.html#L368-L374

So this is just a name thing. 

`output-file` will rename the output file which avoid overriding the other one, but for dependencies, they are tied to a format, and so still stored in original name without removal or being renamed. 

Possibly this could be improved in future Quarto version when will be able to compute a rendering plan which will allow us to know files and resources and correctly name them.

**Reply by perellonieto** [reply]
I have been working in a simpler example in the same repository (https://github.com/perellonieto/quarto_html_revealjs_test)  that shows the problem with two cases and then I saw your answer.

I will still write here me new explanation of the problem which is now in the readme file.

This example includes two `_quarto.yml` configuration files with the only
difference being for case1 `output-dir: .` and for case2 `output-dir: _site`.

The only quarto markdown file to render is `index.qmd` which specifies two
output formats: html and revealjs.

The Makefiles simplifies the task of rendering both examples. 

## Case 1

When rendering the case1 `output-dir: .` results in the correct generation of
two folders for the figures of the html and the revealjs versions inside of
`index_files`. The resulting `index.html` and `slides-index.html` are rendered
correctly when open in a web browser (I use `python -m SimpleHTTPServer` to view it locally).

```
Quarto version 1.3.450
output-dir: .
.
├── index_files
│   ├── figure-html
│   └── figure-revealjs
├── index.html
├── index.qmd
├── Makefile
├── _quarto.yml.case1
├── _quarto.yml.case2
├── README.md
├── requirements.txt
├── search.json
├── site_libs
│   ├── bootstrap
│   ├── clipboard
│   ├── quarto-html
│   ├── quarto-nav
│   ├── quarto-search
│   └── revealjs
└── slides-index.html

10 directories, 9 files
```

## Case 2

When rendering the case2 `output-dir: _site` the `figure-revealjs` folder is
missing. The resulting `slides-index.html` has the figures missing.

```
Quarto version 1.3.450
output-dir: _site
.
├── index.qmd
├── Makefile
├── _quarto.yml.case1
├── _quarto.yml.case2
├── README.md
├── requirements.txt
└── _site
    ├── index_files
    │   └── figure-html
    ├── index.html
    ├── search.json
    ├── site_libs
    │   ├── bootstrap
    │   ├── clipboard
    │   ├── quarto-html
    │   ├── quarto-nav
    │   ├── quarto-search
    │   └── revealjs
    └── slides-index.html

10 directories, 9 files
```

## GitHub quarto-dev/quarto-actions/publish

The Quarto GitHub action publish works as expected, generating all the folders
and files in the `gh-pages` branch.

**Reply by cderv** [reply]
> Quarto version 1.3.450

Can you try latest Quarto 1.4 pre-release ?  I think we may have fix this already... 🤔

**Reply by cderv** [reply]
From a quick test this works for me 
````
C:\USERS\CHRIS\APPDATA\LOCAL\TEMP\QUARTO\QUARTO_HTML_REVEALJS_TEST\_SITE
├───index_files
│   ├───figure-html
│   └───figure-revealjs
└───site_libs
    ├───bootstrap
    ├───clipboard
    ├───quarto-html
    ├───quarto-nav
    ├───quarto-search
    └───revealjs
        ├───dist
        │   └───theme
        │       └───fonts
        │           ├───league-gothic
        │           └───source-sans-pro
        └───plugin
            ├───highlight
            ├───markdown
            ├───math
            ├───notes
            ├───pdf-export
            ├───quarto-line-highlight
            ├───quarto-support
            ├───reveal-menu
            ├───search
            └───zoom
````

using 
````
> bat .\_quarto.yml
───────┬────────────────────────────────────────────────────────────────────────────────────────────────────────────────
       │ File: .\_quarto.yml
───────┼────────────────────────────────────────────────────────────────────────────────────────────────────────────────
   1   │ project:
   2   │   type: website
   3   │   output-dir: _site
   4   │
   5   │ format:
   6   │   html: default
   7   │   revealjs: default
───────┴─────────────────────
````

And latest Quarto 1.4 version

**Reply by perellonieto** [reply]
You are right @cderv the new version 1.4.533 fixes the aforementioned issue. Now the `_site` folder contains both folders and all the images are shown correctly.

```
Quarto version 1.4.533
output-dir: _site
.
├── index.qmd
├── Makefile
├── _quarto.yml.case1
├── _quarto.yml.case2
├── README.md
├── requirements.txt
└── _site
    ├── index_files
    │   ├── figure-html
    │   └── figure-revealjs
    ├── index.html
    ├── search.json
    ├── site_libs
    │   ├── bootstrap
    │   ├── clipboard
    │   ├── quarto-html
    │   ├── quarto-nav
    │   ├── quarto-search
    │   └── revealjs
    └── slides-index.html

11 directories, 9 files
```

**Reply by cderv** [reply]
Awesome! good to know !

**Reply by iagobaapellaniz** [reply]
I just changed the extension 
```
format:
  html: default
  revealsj:
    output-ext: -rjs.html
```
and quarto does the magic!! Awesome, thank you for this amazing tool!!
